js/classes/entities/midground/chad.js

```diff
@@ -233,19 +245,28 @@ class Chad {
 
 
         // Dash action
-        // When not CHAD is not on the ground and his dash is allowed and the user is trying to dash then enter conditional.
-        if (GAME.user.dashing && !this.isOnGround && this.canDash) {
+        // console.log("DASH TIMER: " + this.dashTimer);
+        if (this.groundDashTimer > 0 && this.isOnGround) {
+            this.groundDashTimer -= GAME.clockTick;
+        } else if (this.groundDashTimer <= 0) {
+            this.canGroundDash = true;
+        }
+                    
+        const canDash = ((this.canAirDash && !this.isOnGround) || 
+                        (this.canGroundDash && this.isOnGround));
+
+        if (GAME.user.dashing && canDash) {
```

Comment:
Nice you figured out how to make Chad dash without pressing space first didn't you. My implementation had that issue.